pkg/volume/csi/nodeinfomanager/nodeinfomanager.go

```diff
@@ -441,6 +597,13 @@ func (nim *nodeInfoManager) installDriverToCSINodeInfo(
 		topologyKeys.Insert(k)
 	}
 
+	volumePluginMechanism := csiv1alpha1.VolumePluginMechanismInTree
+	if migratedFunc, ok := nim.migratedDrivers[driverName]; ok {
+		if migratedFunc() {
+			volumePluginMechanism = csiv1alpha1.VolumePluginMechanismCSI
+		}
+	}
+
 	specModified := true
 	statusModified := true
```

Comment:
should these start false and be turned true if we append or modify anything in the loops?

Reply:
since we're only modifying one driver here it doesn't really matter which way we do it. starting `true` and flipping to `false` seems to make the internal `if existingDriver == newDriver` checks a lot cleaner